infra/token-transfer-server/src/lib/transfer.js

```diff
@@ -69,44 +75,23 @@ async function _checkTransferRequest(userId, grantId, amount, transfer = null) {
  * @param amount
  * @returns {Promise<integer>} Id of the transfer request.
  */
-async function enqueueTransfer(userId, grantId, address, amount, ip) {
-  await _checkTransferRequest(userId, grantId, amount)
+async function enqueueTransfer(grantId, address, amount) {
+  const grant = await _checkTransferRequest(grantId, amount)
```

Comment:
nit: there is probably a potential race condition here in case someone would rapid fire requests to enqueueTransfer. We could serialize this call by adding a watchdog variable or locking the grant row in the DB. Alternatively we could let the script that processes the transaction double check (it should anyway) and refuse to execute the transaction.

Reply:
Ahh yea this is something I wanted to handle and add a good test for. Thanks for picking that up!

Reply:
Done!